### ISSUE TITLE
cross-origin error in full res image downloads

### ISSUE THREAD

**Issue by jywarren** [publiclab/mapknitter#211]
I see:

`Uncaught SecurityError: Failed to execute 'texImage2D' on 'WebGLRenderingContext': The cross-origin image at https://s3.amazonaws.com/grassrootsmapping/warpables/251187/DJI_0310.JPG may not be loaded.`

I guess because we're storing images on Amazon... but therefore we need to proxy them through mapknitter.org. :-( 

@icarito, would a simple in-app Rails proxy be okay, do you think?

**Comment by icarito** [comment]
Yes if this is only for our own setup it doesn't make much sense to do in-app - I can proxy thru nginx as we did for the RSS feeds.

**Comment by jywarren** [comment]
OK -- you were able to do so on the same subdomain? No rush on this, i'm just tinkering with it. Thanks!

**Comment by jywarren** [comment]
The images we'll be pulling from will be at URLs like:  "https://s3.amazonaws.com/grassrootsmapping/" ...

**Comment by jywarren** [comment]
Client side code working now in #213. Waiting on proxy to adjust and merge. Thanks!

**Comment by icarito** [comment]
https://github.com/jywarren/web/issues/34#issuecomment-198277342 yes it needs to be in the same subdomain otherwise CORS will give issues.

**Comment by icarito** [comment]
will do

**Comment by jywarren** [comment]
Cool, shall we agree on a format for the proxying? Would it be at something
like https://mapknitter.org/px?url=https:// or would we forward requests
like https://mapknitter.org/px/dir/image.jpg ?
